zebra-state/src/service/finalized_state.rs

```diff
@@ -297,8 +347,50 @@ impl FinalizedState {
                 for orchard_nullifier in transaction.orchard_nullifiers() {
                     batch.zs_insert(orchard_nullifiers, orchard_nullifier, ());
                 }
+
+                // Add new commitments to the the note commitment trees
+                for sprout_note_commitment in transaction.sprout_note_commitments() {
+                    sprout_note_commitment_tree.append(sprout_note_commitment);
+                }
+                for sapling_note_commitment in transaction.sapling_note_commitments() {
+                    sapling_note_commitment_tree
+                        .append(*sapling_note_commitment)
+                        .expect("must work since it was already appended before in the non-finalized state");
+                }
+                for orchard_note_commitment in transaction.orchard_note_commitments() {
+                    orchard_note_commitment_tree
+                        .append(*orchard_note_commitment)
+                        .expect("must work since it was already appended before in the non-finalized state");
+                }
             }
 
+            // Compute the new anchors and index them
+            batch.zs_insert(sprout_anchors, height, sprout_note_commitment_tree.hash());
+            batch.zs_insert(sapling_anchors, height, sapling_note_commitment_tree.root());
+            batch.zs_insert(orchard_anchors, height, orchard_note_commitment_tree.root());
```

Comment:
Ideally all our trees would have the same API to get the root but that's (changing the Sprout one to match the others) not a blocker